### PR TITLE
Feature/html template

### PR DESCRIPTION
**Pull request by frecklescherokee**

### CODE REVIEW

File: assets/js/homepage.js [hunk 1]
```diff
@@ -51,4 +62,181 @@ function getQueryParams(useDiet, useQuery){
...
+
+  return unorderedListEl;
+}
+
+// function to generate an img within a div
+function generateImageEl(i) {
+  let imgEl = document.createElement('img');
+  let divEl = generateMainContentDiv();
+
+  imgEl.src = "https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcTD1kMb0H58I45e9dgLJqEsU7Nc3W1SwoWtpQ&usqp=CAU";
```
Comment (Sha-94) [inline assets/js/homepage.js]: use  ` imgEl.setAttribute('src', '');`

File: assets/js/homepage.js [hunk 2]
```diff
@@ -51,4 +62,181 @@ function getQueryParams(useDiet, useQuery){
...
+        //listItemEl.p = recipes.title[i];
+        //listItemEl.img = recipes.image[i];
+    // append to list
+    listEl.appendChild(listItemEl);
+  }
+  return listEl;
+}
+
+// function to generate recipe list item
+function generateRecipeListItem(i) {
```
Comment (Sha-94) [inline assets/js/homepage.js]: Refactor 'i' to something with more semantic meaning like 'recipe'

File: assets/js/homepage.js [hunk 3]
```diff
@@ -51,4 +62,181 @@ function getQueryParams(useDiet, useQuery){
...
+  //name = "";
+  //console.log("recipes[i].title");
+  //paragraphEl.textContent = name;
+  divEl.appendChild(paragraphEl);
+
+  return divEl;
+}
+
+// function to generate an anchor <a> within a div
+function generateAnchorEl(i) {
```
Comment (Sha-94) [inline assets/js/homepage.js]: You can remove this function, what we can do is add a click listener to the list item element and then generate an new page whenever the element is clicked

File: assets/js/homepage.js [hunk 4]
```diff
@@ -51,4 +62,181 @@ function getQueryParams(useDiet, useQuery){
...
+}
+
+/* Start Page Element Generators */
+
+// function to generate the main page input box
+function generateInputDivEl() {
+    let inputEl = document.createElement('input');
+    let divEl = generateMainContentDiv();
+
+    inputEl.setAttribute('placeholder', 'Search for Recipes');
```
Comment (Sha-94) [inline assets/js/homepage.js:159]: Add id to this element

File: assets/js/homepage.js [hunk 5]
```diff
@@ -51,4 +62,181 @@ function getQueryParams(useDiet, useQuery){
...
+  // append the 3 generated elements to the <main> element
+  mainEl.appendChild(headingDivEl);
+  mainEl.appendChild(inputDivEl);
+  mainEl.appendChild(submitButtonDivEl);
+
+  // create DOM element to reference the fetch/submit button we just added so it can be listened to
+  fetchButtonEl = document.getElementById('fetch-button');
+
+  //event listener for "fetch" button
+  fetchButtonEl.addEventListener('click', fetchRecipes);
```
Comment (Sha-94) [inline assets/js/homepage.js:149]: Move eventListener setup to the generateSubmitButtonDivEl() function

**Review by Sha-94 — CHANGES_REQUESTED** [review]

**Review by Sha-94 — APPROVED** [review]

**Review by daniela-93 — APPROVED** [review]

**Pull request merged**